sql/core/src/test/resources/sql-tests/results/cte.sql.out

```diff
@@ -1,5 +1,5 @@
 -- Automatically generated by SQLQueryTestSuite
--- Number of queries: 27
+-- Number of queries: 63
```

Comment:
I checked these queries in this file work well in postgresql;
rewritten queries for postgresql: https://gist.github.com/maropu/0f7ac12ef3f1b6c6262ecfda4be6be09
output: https://gist.github.com/maropu/a3c12d50058157c6e697ba2a0d4b19dd

Reply:
Thanks. There are a few cases where my implementation allows a bit more than PostgresSQL does, this is because I also tested these queries against MSSQL where they work. I will reply to your findings below one by one.

Reply:
IMHO its ok to support minimal syntaxes for `WITH RECURSIVE` in this pr because the most controversial part in this pr is the implementation of `RecursiveTableExec` about how spark should execute recursive queries. In this pr, it would be better to concentrate on the discussion.

Reply:
All right, then I believe only the most simple case should be allowed in this PR, which is 1 non-recursive term then `UNION ALL` then 1 recursive term within a `WITH RECURSIVE` is allowed.
(And maybe follow-up PRs can deal with advanced constructs.)

Reply:
I simplified the PR, dropped the support for multiple anchor and recursive terms.